client/my-sites/email/form/mailboxes/components/utilities/mailbox-operations.ts

```diff
@@ -0,0 +1,53 @@
+import i18n from 'i18n-calypso';
+import { MailboxForm } from 'calypso/my-sites/email/form/mailboxes';
+import { EmailProvider } from 'calypso/my-sites/email/form/mailboxes/types';
+
+export class MailboxOperations {
+	public mailboxes: MailboxForm< EmailProvider >[];
+	private readonly setMailboxesState: () => void;
+
+	constructor( mailboxes: MailboxForm< EmailProvider >[], setMailboxesState: () => void ) {
+		this.mailboxes = mailboxes;
+		this.setMailboxesState = setMailboxesState;
+	}
+
+	validateLocal() {
+		this.mailboxes.forEach( ( mailbox ) => {
+			mailbox.validate( true );
+		} );
+	}
+
+	async validateForExtraItemsPurchase() {
+		await Promise.all( this.mailboxes.map( ( mailbox ) => mailbox.validateOnDemand() ) );
```

Comment:
Those methods are expected to execute async/remote calls ( often http requests ). It might be best to run them when the user is about to submit to avoid unnecessary calls to the backend.

We could look at it after, but my initial intention was to ensure we had the existing functionality.